newIDE/app/src/EventsSheet/EventsTree/index.js

```diff
@@ -128,80 +130,92 @@ type EventsContainerProps = {|
+    props.onEventContextMenu(domEvent.clientX, domEvent.clientY);
   };
 
-  render() {
-    const { event, project, scope, disabled } = this.props;
-    const EventComponent = EventsRenderingService.getEventComponent(event);
+  const longTouchForContextMenuProps = useLongTouch(
+    React.useCallback(
+      (domEvent: any) => {
+        onEventContextMenu(domEvent.clientX, domEvent.clientY);
+      },
+      [onEventContextMenu]
+    )
+  );
 
-    return (
-      <div
-        ref={container => (this._container = container)}
-        onClick={this.props.onEventClick}
-        onContextMenu={this._onEventContextMenu}
-      >
-        {!!EventComponent && (
-          <div style={styles.eventComponentContainer}>
-            {this.props.connectDragSource(<div className={handle} />)}
-            <div style={styles.container}>
-              <EventComponent
-                project={project}
-                scope={scope}
-                event={event}
-                globalObjectsContainer={this.props.globalObjectsContainer}
-                objectsContainer={this.props.objectsContainer}
-                selected={isEventSelected(this.props.selection, event)}
-                selection={this.props.selection}
-                leftIndentWidth={this.props.leftIndentWidth}
-                onUpdate={this._onEventUpdated}
-                onAddNewInstruction={this.props.onAddNewInstruction}
-                onPasteInstructions={this.props.onPasteInstructions}
-                onMoveToInstruction={this.props.onMoveToInstruction}
-                onMoveToInstructionsList={this.props.onMoveToInstructionsList}
-                onInstructionClick={this.props.onInstructionClick}
-                onInstructionDoubleClick={this.props.onInstructionDoubleClick}
-                onInstructionContextMenu={this.props.onInstructionContextMenu}
-                onAddInstructionContextMenu={
-                  this.props.onAddInstructionContextMenu
-                }
-                onEndEditingEvent={this.props.onEndEditingEvent}
-                onParameterClick={this.props.onParameterClick}
-                onOpenExternalEvents={this.props.onOpenExternalEvents}
-                onOpenLayout={this.props.onOpenLayout}
-                disabled={
-                  disabled /* Use disabled (not event.disabled) as it is true if a parent event is disabled*/
-                }
-                renderObjectThumbnail={this.props.renderObjectThumbnail}
-                screenType={this.props.screenType}
-                eventsSheetHeight={this.props.eventsSheetHeight}
-                windowWidth={this.props.windowWidth}
-              />
-            </div>
+  const EventComponent = EventsRenderingService.getEventComponent(event);
+
+  return (
+    <div
+      ref={containerRef}
+      onClick={props.onEventClick}
+      onContextMenu={_onEventContextMenu}
+      {...longTouchForContextMenuProps}
```

Comment:
Note: the content of `longTouchForContextMenuProps` is stable across render (because everything inside is useCallback'ed and the callback of useLongTouch is itself useCallback'ed), so if _onEventContextMenu is, then as long as onEventClick is stable, the whole div is stable.
Not super important, but one less DOM element for React to update at every render :)